ci-operator/step-registry/ovn/sdn-migration/feature-migration/validate-egress-firewall/ovn-sdn-migration-feature-migration-validate-egress-firewall-commands.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+set -x
+set -o errexit
+set -o nounset
+set -o pipefail
+
+validate_egressfirewall_cr () {
+  oc get egressfirewll -n test-migration default -o=jsonpath='{.spec}'>current_config
+  echo "$EXPECT_EGRESS_FIREWALL_SPEC" > expected_config
+  if diff <(jq -S . current_config) <(jq -S . expected_config); then
+    echo "configuration is migrated as expected"
+  else
+    echo "configuration is not migrated as expected"
+  fi
+}
+
+validate_egressnetworkpolicy_cr () {
+  oc get egressnetworkpolicy -n test-migration default -o=jsonpath='{.spec}'>current_config
+  echo "$EXPECT_EGRESS_FIREWALL_SPEC" > expected_config
+  if diff <(jq -S . current_config) <(jq -S . expected_config); then
+    echo "configuration is migrated as expected"
+  else
+    echo "configuration is not migrated as expected"
+  fi
+}
+
+TMPDIR=$(mktemp -d)
+pushd ${TMPDIR}
+
+echo "check the cluster running CNI"
+RUNNING_CNI=$(oc get network.operator cluster -o=jsonpath='{.spec.defaultNetwork.type}')
+
+if [[ $RUNNING_CNI == "OpenShiftSDN" ]]; then
```

Comment:
Just a note for myself: This check works fine for EgressFirewall. 

For my EgressIP migration's CI though, the timing may be more stringent for when resources are generated in forward and rollback migration. I will check against defaultNetworkType, but also MigrationNetworkType shall be "" (empty string).